jax/experimental/maps.py

```diff
@@ -166,20 +166,33 @@ def fresh_resource_name(tag=None):
   finally:
     _next_resource_id += 1
 
+
 # This is really a Dict[AxisName, int], but we don't define a
 # pytree instance for it, so that it is treated as a leaf.
 class AxisNamePos(FrozenDict):
-  pass
+  user_repr: Any
+
+  def __init__(self, *args, user_repr, **kwargs):
+    super().__init__(*args, **kwargs)
+    self.user_repr = user_repr
+
+
+# str(...) == 'Ellipsis' which is really annoying
+class DotDotDotRepr:
+  def __repr__(self): return '...'
```

Comment:
If `user_repr` above is always a string (as it is at the only two current call sites), instead of having this class we could just have a literal `'...'` where the class constructor is currently called. Maybe you have some generalization in mind though, where it'd be handy to have these objects in tuples or lists (and maybe `user_repr` isn't always a string)?

Reply:
Replacing `user_repr=str(entry[:-1] + [DotDotDotRepr()])` with `user_repr=str(entry[:-1] + ['...'])`would change `['i', 'j', ...]` (which is what the user writes) with `['i', 'j', '...']` which might be a little confusing. I use the dummy class just to avoid the extra quotes.

Reply:
I was thinking that it'd be outside the `str` call rather than adding it to the list, like `user_repr=f"[{', '.join(map(repr, entry[:-1]))}, ...]"`. Now that I write it out it's not obviously simpler than having the class, though it is a bit more local.

Anyway, I don't mean to bikeshed! Anything LGTM.